src/obr/core/queries.py

```diff
@@ -244,3 +298,42 @@ def query_to_dataframe(
     if index:
         return ret.set_index(index).sort_index()
     return ret
+
+
+def build_filter_query(filters: Union[list[str], tuple[str]]) -> list[Query]:
+    """This function builds a list of filter queries, where filter queries are queries that request a specific value and has to conform a predicate"""
+    q: list[Query] = []
+
+    # query syntax
+    if not any([pred for pred in Predicates if pred.value in filters[0]]):
```

Comment:
Having trouble wrapping my head around this block. Could you expand the comment a bit more to make it easier to understand what is happening here.

Reply:
Will do!